### DISCUSSION TITLE
Validation on path parameters?

### DISCUSSION THREAD

**Discussion by epotter2297** [ecyrbe/zodios#176] [Ideas]
It would be really cool if there was a way to use zod to validate my path parameters. For example, if I have an endpoint like so:
```js
{
    method: 'get',
    path: '/users/:id',
    alias: 'getUser',
    description: 'Get a user',
    response: z.object({
      id: z.number(),
      name: z.string().nullable(),
    }),
},
```
And I want to make sure that `id` is a valid UUID, I could do something like:
```js
{
    method: 'get',
    path: '/users/:id',
    alias: 'getUser',
    description: 'Get a user',
    parameters: [
      {
        name: 'id',
        description: 'The user id',
        type: 'Path',
        schema: z.string().uuid(),
      },
    ],
    response: z.object({
      id: z.number(),
      name: z.string().nullable(),
    }),
},
```

**Comment by ecyrbe** [comment]
Hello,

Thank you for asking.
This could be possible. However i'd like to also keep path parameters inference feature zodios has today.
So this should be completely optional to validate path parameters.
Can you create a feature request ?

**Comment by ecyrbe** [comment]
nevermind, there is a button to create a feature request from discussion

**Comment by ecyrbe** [comment]
@epotter2297 it is now available on
- `@zodios/core` v10
- `@zodios/express` v10

**Reply by epotter2297** [reply]
Awesome, thank you! 

Just took a quick look and it does exactly what I was thinking 🙂 

I do see that in my router `req.params.id` still has a type of `string | number` and it would be cool if it could infer the type I specified earlier, but this is already a good improvement as-is.

**Comment by ecyrbe** [comment]
Indeed, i created an issue #180 to track this

**Comment by ecyrbe** [comment]
@epotter2297  better path param type narrowing is now available on:  
- `@zodios/core` v10.0.1
- `@zodios/express` v10.0.1

**Reply by epotter2297** [reply]
Okay, we're nearly there now 🙂 

The one issue I see now is that while `req.params.id` correctly infers the type as `number`, when I actually hit the endpoint `id` is still a string at runtime.

I can open an issue for it quick.
